lib/server.js

```diff
@@ -1170,22 +1173,24 @@ Server.prototype._setupRequest = function _setupRequest(req, res) {
         req._timeUseEnd = req._timeUseEnd || processHrTime;
         req._timeRouteEnd = req._timeRouteEnd || processHrTime;
 
-        req.removeListener('close', onClose);
-        res.removeListener('error', onError);
-
-        // Do not trigger false
         self._finishReqResCycle(req, res);
     }
-    function onError(err) {
+
+    // Emitted when the response has been sent.
+    function onResClose() {
         res._flushed = true;
-        req._timeFlushed = process.hrtime();
+        // Finish may alread set flushed timer
+        req._timeFlushed = req._timeFlushed || process.hrtime();
 
-        req.removeListener('close', onClose);
-        res.removeListener('finish', onFinish);
-        self._finishReqResCycle(req, res, err);
+        self._finishReqResCycle(req, res, res.err);
```

Comment:
Why do we need to call `_finishReqResCycle` here?

Reply:
To tigger after (after is triggered after both handlers are finished and req is flushed)